assets/app/view/game/round/auction.rb

```diff
@@ -68,6 +68,13 @@ def render_show_button
             if @block_show
               store(:flash_opts, 'Enter master mode to reveal other hand. Use this feature fairly.')
             else
+              if @current_actions.include?('pass')
+                @step.before_process_pass = lambda do
```

Comment:
hm i'm not sure about this. is there another way? it seems weird for the client to mutate the engine like this

Reply:
Yeah, this is definitely super weird. Typing up a comment got me thinking more about this, and I had an idea for keeping this `before_process_pass` stuff only under `assets/app/view/` instead of in the engine -- https://github.com/tobymao/18xx/pull/1656/commits/7ee232ade161c56a9bb6083155afe33a6c8e33dc